lib/postgrex/type_supervisor.ex

```diff
@@ -3,29 +3,57 @@ defmodule Postgrex.TypeSupervisor do
 
   use Supervisor
 
+  @manager Postgrex.TypeManager
+  @servers Postgrex.TypeSupervisor
+
+  @doc """
+  Starts a type supervisor with a manager and a simple
+  one for one supervisor for each server.
+  """
   def start_link(mode \\ :manager)
 
   def start_link(:manager) do
     Supervisor.start_link(__MODULE__, :manager)
   end
+
   def start_link(:servers) do
-    Supervisor.start_link(__MODULE__, :servers, [name: __MODULE__])
+    Supervisor.start_link(__MODULE__, :servers, name: @servers)
+  end
+
+  @doc """
+  Locates a type server for the given module-key pair.
+  """
+  def locate(module, key) do
+    pair = {module, key}
+
+    case Registry.lookup(@manager, pair) do
+      [{pid, _}] -> if Process.alive?(pid), do: pid, else: start_server(module, pair)
```

Comment:
Do we really gain anything by this check? It still does not give any guarantees, since the process might die right after this check.

Reply:
I am reproducing the logic of `whereis_name` which does check if the process is alive. The registry is not atomic regarding DOWN messages, so this is necessary.

Reply:
Since we already trap exits in the type server process, I wonder if it would make sense to try to "mark dying" the server in the registry from the terminate callback.

Reply:
I also think that `Process.alive?/1` is unnecessary here.